merkle/src/merkle.rs

```diff
@@ -98,7 +96,11 @@ pub trait Store<E: Element>:
     /// Creates a new store which can store up to `size` elements.
     // FIXME: Return errors on failure instead of panicking
     //  (see https://github.com/filecoin-project/merkle_light/issues/19).
-    fn new(size: usize) -> Self;
+    fn new(size: usize) -> Self {
+        Self::new_with_file(size, None)
+    }
+
+    fn new_with_file(size: usize, file: Option<File>) -> Self;
```

Comment:
Lets use this occasion to make all these return `Result<Self>`, we really need to handle errors here when we try caches etc

Reply:
@dignifiedquire Should it use `rust-fil-proofs`s MT defined errors like `MerkleTreeGenerationError` or define its own (and `rust-fil-proofs` import those)?

Reply:
you can either stick with `io::Error` or if those are not enough, use `failure` to define its own, and then add a `MerkleTree` variant in rust-fil-proofs to absorb these

Reply:
Added `Result<Self>` for the file-related API.